### PR TITLE
Routing with Apollo middlewares instead of configurable RCHC-utility-function

### PR DESCRIPTION
**Pull request by nikomakela**
HH-120

> The use of the utility functions like getI18nPath and getRoutedInternalHref is a bad practise, because they need to be used in every URL and every link all around the application and also in the external plugins, which makes it really hard to maintain and to understand. The middleware approach would be a much cleaner solution. However, it is hard to transform just the right set of the links and URLs of the response documents in the middleware. The Headless CMS should at least give a hint in the URL about the context domain that the links represent: The link could always include the domain as a path in the URL, e.g `https://hobbies.cms/articles/[uri|en/category/slug]` and `https://hobbies.cms/pages/[uri|en/slug]`, which would make it easy for middleware to just search and replace the link contents to reroute them internally.

### CODE REVIEW

File: src/domain/app/AppConfig.ts [hunk 1]
```diff
@@ -94,6 +94,19 @@ class AppConfig {
   static get showSimilarEvents() {
     return Boolean(parseEnvValue(process.env.NEXT_PUBLIC_SHOW_SIMILAR_EVENTS));
   }
+
+  static get URLTransformMapping() {
+    return {
+      'https://api.hel.fi/linkedevents/v1/event': '/courses',
```
Comment (nenonja) [inline src/domain/app/AppConfig.ts:100]: these domains probably should be read from env configuration?

Comment (nikomakela) [reply src/domain/app/AppConfig.ts:100]: yea and they should not be that hard coded at all. I mean, we cannot have hard coded url routes for articles categories for example. They are too dynamic. 

This is the reason why this PR is still in draft mode. We need more specific URL's from the Headless CMS. The context needs to be told in the URL.

File: src/hooks/useRHHCConfig.tsx [hunk 2]
```diff
@@ -43,48 +39,16 @@ export default function useRHHCConfig(
...
-    };
     return {
       ...rhhcDefaultConfig,
       components: {
         ...rhhcDefaultConfig.components,
         Head: (props) => <Head {...props} />,
         Link: ({ href, ...props }) => <Link href={href || ''} {...props} />,
-        EventCardContent: (props) => <EventDetails {...props} />,
-        ArticleCardContent: (props) => <ArticleDetails {...props} />,
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (nenonja) [inline src/hooks/useRHHCConfig.tsx:48]: Could this work with type of `Record<string,unknown>` or just`unknown` ?

Comment (nikomakela) [reply src/hooks/useRHHCConfig.tsx:48]: Actually, I think they should be typed in the HCRC-lib. Some reason why the typing is not exported after the build done by the Rollup.

Comment (nikomakela) [reply src/hooks/useRHHCConfig.tsx:48]: But yeah, the unknown would be better than any with disabled linter. 👍

Comment (nenonja) [reply src/hooks/useRHHCConfig.tsx:48]: https://thoughtbot.com/blog/typescript-stop-using-any-there-s-a-type-for-that

File: src/utils/routerUtils.ts [hunk 3]
```diff
@@ -173,8 +173,23 @@ export function getLocalizedCmsItemUrl(
...
+    locale !== DEFAULT_LANGUAGE ? `/${locale}` : ''
+  }${stringifyUrlObject({
     query: query,
     pathname: getI18nPath(pathname, locale),
   })}`;
 }
+
+export function transformResponsesInternalURLs(
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  apolloResponseData: Record<string, any>
```
Comment (nenonja) [inline src/utils/routerUtils.ts:186]: How about `Record<string, unknown>` ?

**Review by nenonja — COMMENTED** [review]
Some comments left. Also:
- Pr has some conflicts
- some unit or snapshot tests could be added for `transformResponsesInternalURLs` function with "real" data taken from cms and linkedevents responses.

**Review by melniiv — APPROVED** [review]

**Pull request merged**